apps/src/codebridge/FileBrowser/FileRow.tsx

```diff
@@ -0,0 +1,139 @@
+  appName?: string;
+  hasValidationFile: boolean; // If the project has a validation file already.
+  isStartMode: boolean;
+  renameFilePrompt: renameFilePromptType;
+  handleDeleteFile: (fileId: string) => void;
+  setFileType: setFileType;
+}
+
+const handleFileDownload = (file: ProjectFile, appName: string | undefined) => {
+  fileDownload(file.contents, file.name);
+  sendCodebridgeAnalyticsEvent(EVENTS.CODEBRIDGE_DOWNLOAD_FILE, appName);
+};
+
+/**
+ * A single file row in the file browser. This component does not handle
+ * drag and drop, that is handled by the parent component.
+ */
+const FileRow: React.FunctionComponent<FileRowProps> = ({
+  file,
+  isReadOnly,
+  enableMenu,
+  appName,
+  hasValidationFile,
+  isStartMode,
+  renameFilePrompt,
+  handleDeleteFile,
+  setFileType,
+}) => {
+  const {
+    openFile,
+    config: {editableFileTypes},
+  } = useCodebridgeContext();
+  const {openMoveFilePrompt} = usePrompts();
+  const {iconName, iconStyle, isBrand} = getFileIconNameAndStyle(file);
+  const iconClassName = isBrand
+    ? classNames('fa-brands', moduleStyles.rowIcon)
+    : moduleStyles.rowIcon;
+  const isLocked = !isStartMode && file.type === ProjectFileType.LOCKED_STARTER;
+
+  return (
+    <div className={moduleStyles.row}>
+      <div className={moduleStyles.label} onClick={() => openFile(file.id)}>
+        <FontAwesomeV6Icon
+          iconName={iconName}
+          iconStyle={iconStyle}
+          className={iconClassName}
+        />
+
+        <OverflowTooltip
+          tooltipProps={{
+            text: file.name,
+            tooltipId: `file-tooltip-${file.id}`,
+            size: 's',
+            direction: 'onBottom',
+          }}
+          tooltipOverlayClassName={moduleStyles.nameContainer}
+          className={moduleStyles.nameContainer}
+        >
+          <span>{file.name}</span>
+        </OverflowTooltip>
+      </div>
+      {!isReadOnly && enableMenu && (
+        <PopUpButton
+          iconName="ellipsis-v"
+          className={moduleStyles['button-kebab']}
+        >
+          <span className={moduleStyles['button-bar']}>
+            {!isLocked && (
+              <PopUpButtonOption
+                iconName="arrow-right"
+                labelText={codebridgeI18n.moveFile()}
+                clickHandler={() => openMoveFilePrompt({fileId: file.id})}
+              />
+            )}
+            {!isLocked && (
+              <PopUpButtonOption
+                iconName="pencil"
+                labelText={codebridgeI18n.renameFile()}
+                clickHandler={() => renameFilePrompt(file.id)}
+              />
+            )}
+            {editableFileTypes.some(type => type === file.language) && (
+              <PopUpButtonOption
+                iconName="download"
+                labelText={codebridgeI18n.downloadFile()}
+                clickHandler={() => handleFileDownload(file, appName)}
+              />
+            )}
+            {!isLocked && (
```

Comment:
Noticing how often we repeat the `!isLocked && (<PopUpButtonOption ../>` logic, I wonder about a refactor like this (mainly because I like the declarative logic of a `fileOptions` array having the configuration for each option) but feel free to take or leave. 

```
const FileOptions = ({ file, isLocked, editableFileTypes, appName, isStartMode, hasValidationFile, setFileType }) => {
  const fileOptions = [
    {
      condition: !isLocked,
      iconName: "arrow-right",
      labelText: codebridgeI18n.moveFile(),
      clickHandler: () => openMoveFilePrompt({ fileId: file.id }),
    },
    {
      condition: !isLocked,
      iconName: "pencil",
      labelText: codebridgeI18n.renameFile(),
      clickHandler: () => renameFilePrompt(file.id),
    },
    {
      condition: editableFileTypes.includes(file.language),
      iconName: "download",
      labelText: codebridgeI18n.downloadFile(),
      clickHandler: () => handleFileDownload(file, appName),
    },
    {
      condition: !isLocked,
      iconName: "trash",
      labelText: codebridgeI18n.deleteFile(),
      clickHandler: () => handleDeleteFile(file.id),
    }
  ];

  return (
    <>
      {fileOptions.map(({ condition, iconName, labelText, clickHandler }, index) => 
        condition && (
          <PopUpButtonOption
            key={index}
            iconName={iconName}
            labelText={labelText}
            clickHandler={clickHandler}
          />
        )
      )}
      {isStartMode && (
        <StartModeFileDropdownOptions
          file={file}
          projectHasValidationFile={hasValidationFile}
          setFileType={setFileType}
        />
      )}
    </>
  );
};
```

Reply:
updated!